### PR TITLE
add io.github.rocket.editor

### PR DESCRIPTION
**Pull request by kusma**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project.
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: io.github.rocket.editor.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    </screenshot>
+  </screenshots>
+
+  <url type="homepage">https://rocket.github.io/</url>
+
+  <content_rating type="oars-1.1" />
+
+  <releases>
+    <release version="0.11.0" date="2018-08-01" />
+  </releases>
```
Comment (kusma) [inline io.github.rocket.editor.appdata.xml:34]: It's a bit unclear to me if the releases should be *all* releases ever made (including source-only releases), or just the ones for this flatpak. If it's the former, I guess I'm missing a few here. If it's the latter, the question becomes if the date should be the date of the source-release or the flatpak release. Those differ quite widely in this case...

Comment (JakobDev) [reply io.github.rocket.editor.appdata.xml:34]: It only requires the latest release. You can add former if you want, so Software Centers can show a full changelog, but it is not required.

**Comment by JakobDev** [comment]
You should upstream the AppStream, the Desktop Entry and the Icon.

**Comment by JakobDev** [comment]
bot, build io.github.rocket.editor

File: io.github.rocket.editor.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+app-id: io.github.rocket.editor
+runtime: org.kde.Platform
+runtime-version: '5.15-22.08'
+sdk: org.kde.Sdk
+command: editor
+finish-args:
+  - --socket=wayland
```
Comment (JakobDev) [inline io.github.rocket.editor.yml:7]:
```suggestion
  - --socket=wayland
  - --socket=fallback-x11
```
Or else it would not work under X11

**Comment by kusma** [comment]
> Error: Command 'editor' not found

Hmm, that's what I got when I didn't patch the install target. With that patch, it works fine locally. I wonder what's going on...

**Comment by kusma** [comment]
> You should upstream the AppStream, the Desktop Entry and the Icon

Yes, I intend to. The upstream project isn't in a great state right now, with our CI being broken. I'm working on that right now.

File: io.github.rocket.editor.appdata.xml (hunk 1 above)
Comment (hfiguiere) [reply io.github.rocket.editor.appdata.xml:34]: it's always when upstream released. I usually take the tag date if I have to guess, or the tarball date.

File: io.github.rocket.editor.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  - --socket=fallback-x11
+  - --share=network
+  - --device=dri
```
Comment (hfiguiere) [inline io.github.rocket.editor.yml:10]: `- --share=ipc` is also needed.

File: io.github.rocket.editor.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      - type: file
+        path: io.github.rocket.editor.svg
+      - type: file
+        path: io.github.rocket.editor.desktop
+      - type: file
+        path: io.github.rocket.editor.appdata.xml
+    build-commands:
+      - install -Dm644 -t /app/share/icons/hicolor/scalable/apps io.github.rocket.editor.svg
+      - install -Dm644 -t /app/share/applications/ io.github.rocket.editor.desktop
+      - install -Dm644 -t /app/share/metainfo io.github.rocket.editor.appdata.xml
```
Comment (hfiguiere) [inline io.github.rocket.editor.yml:34]:
```suggestion
    post-install:
      - install -Dm644 -t /app/share/icons/hicolor/scalable/apps ../io.github.rocket.editor.svg
      - install -Dm644 -t /app/share/applications/ ../io.github.rocket.editor.desktop
      - install -Dm644 -t /app/share/metainfo ../io.github.rocket.editor.appdata.xml
    sources:
      - type: git
        url: https://github.com/rocket/rocket.git
        tag: v0.11.0
      - type: patch
        path: 0001-install-editor.patch
        use-git: true
      - type: file
        path: io.github.rocket.editor.svg
      - type: file
        path: io.github.rocket.editor.desktop
      - type: file
        path: io.github.rocket.editor.appdata.xml
```

Comment (kusma) [reply io.github.rocket.editor.yml:34]: Yeah, that seems nicer!

File: io.github.rocket.editor.yml (hunk 3 above)
Comment (kusma) [reply io.github.rocket.editor.yml:11]: Applied that change, and squashed it all! Thanks!

**Comment by kusma** [comment]
> > Error: Command 'editor' not found
> 
> Hmm, that's what I got when I didn't patch the install target. With that patch, it works fine locally. I wonder what's going on...

Just tried again, and I can't reproduce it. Anyone able to trigger another build and see if it still happens?

**Comment by kusma** [comment]
It's really strange.

```
make: Nothing to be done for 'install'.
```

This seems to indicate that the patching didn't work; there's nothing in the install target.

But I see the patching work:

```
Applying patch 0001-install-editor.patch
FB: Running: git apply -v -p1 /srv/buildbot/worker/build-x86_64-2/build/0001-install-editor.patch
Checking patch editor/editor.pro...
Applied patch editor/editor.pro cleanly.
```

I'm really lost as to what's going wrong here...

**Comment by hfiguiere** [comment]
the previous failed build say `make: Nothing to be done for 'install'.`

You can still install it in the post-install. That should be enough.

**Comment by kusma** [comment]
bot, build io.github.rocket.editor

File: io.github.rocket.editor.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      - install -Dm644 -t /app/share/metainfo ../io.github.rocket.editor.appdata.xml
+    sources:
+      - type: git
+        url: https://github.com/rocket/rocket.git
+        tag: v0.11.0
+      - type: patch
+        path: 0001-install-editor.patch
+        use-git: true
+      - type: file
+        path: io.github.rocket.editor.svg
```
Comment (hfiguiere) [inline io.github.rocket.editor.yml:28]: you should be able to just use the file in `contrib/logo.svg` instead.

**Comment by kusma** [comment]
@hfiguiere: Yeah, that's a fair point. Maybe that's a bit less error-prone in the first place.

File: io.github.rocket.editor.yml (hunk 5 above)
Comment (kusma) [reply io.github.rocket.editor.yml:28]: Yeah, that's the same file, good point.

**Comment by hfiguiere** [comment]
also is it supposed to play sounds? I see no permission for it if that's the case.

**Comment by kusma** [comment]
> also is it supposed to play sounds? I see no permission for it if that's the case.

No, it doesn't play sound by itself. That's up to the "remote end", which is connected over a socket, and something that I don't control.

**Comment by kusma** [comment]
bot, build io.github.rocket.editor

File: io.github.rocket.editor.yml (hunk 5 above)
Comment (kusma) [reply io.github.rocket.editor.yml:27]: OK, done.

**Comment by kusma** [comment]
bot, build io.github.rocket.editor

**Comment by kusma** [comment]
OK, seems close. The built version works fine here, but: the screenshot doesn't load for me any more. It did before, so I'm not quite sure what's up with that. Is this maybe some optimization for non-production builds that's expected?

**Comment by kusma** [comment]
With the local build, the screenshot works fine...

**Comment by kusma** [comment]
Tested the aarch64 build as well, and it seems to work just fine!

**Pull request closed without merging**